crates/bevy_ecs/src/world/mod.rs

```diff
@@ -1207,29 +1194,26 @@ impl World {
     #[inline]
     unsafe fn initialize_resource_internal(&mut self, component_id: ComponentId) -> &mut Column {
         // SAFE: resource archetype always exists
-        let resource_archetype = self
-            .archetypes
-            .archetypes
-            .get_unchecked_mut(ArchetypeId::RESOURCE.index());
-        let resource_archetype_components = &mut resource_archetype.components;
+        let resources = &mut self.storages.resources;
+        let resources = &mut resources.resources;
```

Comment:
```suggestion
        let resources = &mut self.storages.resources.resources;
```

Reply:
Oh man just realized how hard that is to read with the double resources in there.